pages/learn/miscellaneous-examples.md

```diff
@@ -157,3 +173,291 @@ We also introduce the following with this example:
+    },
+    "hobbies": {
+      "type": "array",
+      "items": {
+        "type": "string"
+      }
+    }
+  },
+  "required": ["name", "age"]
+}
+```
+
+**Data**
+
+```json
+{
+  "name": "John Doe",
+  "age": 25,
+  "address": {
+    "street": "123 Main St",
+    "city": "New York",
+    "state": "NY",
+    "postalCode": "10001"
+  },
+  "hobbies": ["reading", "running"]
+}
+```
+
+The provided data conforms to the schema by including values for the required properties and ensuring the `age` is an integer greater than or equal to zero. The `address` object contains all the necessary properties, and the `hobbies` property is an array of strings.
+
+
+## Conditional validation with dependentRequired
+
+In this example, the `dependentRequired` keyword is used to specify that the property `bar` is required when the property `foo` is present. The schema enforces the condition that if `foo` exists, then `bar` must also be present.
+
+```json
+{
+  "$id": "https://example.com/conditional-validation-dependentRequired.schema.json",
+  "$schema": "https://json-schema.org/draft/2020-12/schema",
+  "title": "Conditional Validation with dependentRequired",
+  "type": "object",
+  "properties": {
+    "foo": {
+      "type": "boolean"
+    },
+    "bar": {
+      "type": "string"
+    }
+  },
+  "dependentRequired": {
+    "foo": ["bar"]
+  }
+}
+```
+
+**Data**
+
+```json
+{
+  "foo": true,
+  "bar": "Hello World"
+}
+```
+
+As per the schema, when the `foo` property is present (`true`), the `bar` property becomes required. The `bar` property is provided with the value "Hello World", satisfying the requirement of being a string and ensuring compliance with the `dependentRequired` condition.
+
+
+## Conditional validation with dependentSchemas
+
+The given schema showcases the use of the `dependentSchemas` keyword. It allows defining a subschema that must be satisfied if a certain property is present. 
+
+* In this example, the schema defines an object with two properties: `foo` and `propertiesCount`. The `foo` property is of boolean type, while the `propertiesCount` property is of integer type with a minimum value of 0.
+* According to the subschema, when the `foo` property is present, the `propertiesCount` property becomes required, and must be an integer with a minimum value of 7.
+
+```json
+{
+  "$id": "https://example.com/conditional-validation-dependentSchemas.schema.json",
+  "$schema": "https://json-schema.org/draft/2020-12/schema",
+  "title": "Conditional Validation with dependentSchemas",
+  "type": "object",
+  "properties": {
+    "foo": {
+      "type": "boolean"
+    },
+    "propertiesCount": {
+      "type": "integer",
+      "minimum": 0
+    }
+  },
+  "dependentSchemas": {
+    "foo": {
+      "required": ["propertiesCount"],
+      "properties": {
+        "propertiesCount": {
+          "type": "integer",
+          "minimum": 7
+        }
+      }
+    }
+  }
+}
+```
+
+**Data**
+
+```json
```

Comment:
It might be good to show an invalid instance where `propertiesCount` is not an integer as well as a valid instance where both `foo` is missing and `propertiesCount` is not an integer, and explanations why.

Reply:
I mistakenly set the `propertiesCount` type as an integer under the `dependentSchemas` again. 😅 
The type is an integer only. However, for the invalid instance, I can set it to be less than 7 instead of the minimum condition of 7. Similarly, for the valid instance where `foo` is missing, I can also set it to be less than 7 since the minimum value should be 0 in that case. 

Please let me know if this works.

Reply:
Ah, yes. I missed the integer declaration up higher.

This sounds good. Thanks.

Reply:
Yes, okay! Will just do it.